rust/trace/src/sys_pid.rs

```diff
@@ -0,0 +1,30 @@
+use libc;
+
+#[cfg(all(target_family = "unix", not(target_os = "fuchsia")))]
+#[inline]
+pub fn current_pid() -> u64 {
+    extern {
+        fn getpid() -> libc::pid_t;
+    }
+
+    unsafe {
+        getpid() as u64
+    }
+}
+
+#[cfg(target_os = "fuchsia")]
+pub fn current_pid() -> u64 {
+    // TODO: implement for fuchsia
```

Comment:
I'm pretty sure this won't compile on fuchsia. You can use `unimplemented!()` or just return a dummy value though.

Reply:
(`unimplemented!()` will compile but panic if called, so maybe not the best)

Reply:
I think returning 0 is the best idea for now, having a compile error would create extra difficulty (would need to be patched in the local mirror).